classes/mail/MailTemplate.inc.php

```diff
@@ -231,6 +231,18 @@ function replaceParams() {
 
 			$subject = str_replace('{$' . $key . '}', $value, $subject);
 		}
+
+		// Clear all params that are not yet assigned
+		preg_match_all('/\{\$.*?\}/', $body, $allUnassignedParams);
+
+		if ($allUnassignedParams != null && sizeof($allUnassignedParams) > 0) {
+			$matches = $allUnassignedParams[0];
+
+			foreach ($matches as $key => $value) {
+				$body = str_replace($value, '', $body);
+			}
+		}
+		
```

Comment:
Can you say more about where you're encountering this issue? My first thought is that we should leave any untransformed `{$something}` text in, because this is a good way to highlight what is missing. 

It may also give some indication to the recipient what has gone wrong. Consider:

> Please submit your revisions at the link below.
> 
> {$revisionsLink}
>
> Thanks for submitting to [Journal of Public Knowledge](...).

With this:

> Please submit your revisions here:
>
>
>
> Thanks for submitting to [Journal of Public Knowledge](...).

In the second case, a recipient might confuse the journal link with the link they are meant to click.

Reply:
Good point. I will remove that. I have to put the `$unsubscribeUrl` somehow automatically, and not as a parameter.

Reply:
Maybe we can just add it to the email templates on upgrade? People can choose to edit it away or place it differently in the email.

Reply:
I added it there, but the notification email template is send independently of whether the `{unsubscribeUrl}` is populated. So it will remain the parameter pattern on the send email, if for example a discussion notification email is send.

Reply:
Sorry, I'm not sure I understand what you're saying here.